x-pack/plugin/frozen-indices/src/internalClusterTest/java/org/elasticsearch/index/engine/FrozenIndexIT.java

```diff
@@ -97,4 +103,73 @@ public void testTimestampRangeRecalculatedOnStalePrimaryAllocation() throws IOEx
         assertThat(timestampFieldRange.getMax(), equalTo(Instant.parse("2010-01-06T02:03:04.567Z").getMillis()));
     }
 
+    public void testTimestampFieldTypeExposedByAllIndicesServices() throws Exception {
+        internalCluster().startNodes(between(2, 4));
+
+        final String locale;
+        final String date;
+
+        final boolean isJdk8 = JavaVersion.current().getVersion().get(0) == 8; // JDK8 doesn't reliably know French
```

Comment:
@pgomulka I've asked for you to take a look at this backport because this test is having trouble with locales and older JDK versions and I think you have expertise in this area. As far as I can tell, the behaviour difference is in JDK8, hence this line, but it'd be good if you could give it a quick look and see if there are any other gotchas here.

The rest of the PR is already reviewed, it's just this one test that needs attention.

Reply:
things to bear in mind is what locales are supported with `-Dlocale.providers=COMPAT` in newer jdks. French is a safe choice. For instance here `https://www.oracle.com/java/technologies/javase/java9locales.html` see `The locales supported by the COMPAT provider are as follows.`

I think it should be ok to use jdk8 same way as jkd9+. If it fails we could look into a reproduction line. 
But also agree comparing jdk behaviour is not the point of this test .